packages/server/src/sdk/app/tables/migration.ts

```diff
@@ -1,53 +1,65 @@
-import { BadRequestError, context } from "@budibase/backend-core"
+import { BadRequestError, context, db as dbCore } from "@budibase/backend-core"
 import {
   BBReferenceFieldMetadata,
   FieldSchema,
   FieldSubtype,
   InternalTable,
-  ManyToManyRelationshipFieldMetadata,
-  ManyToOneRelationshipFieldMetadata,
-  OneToManyRelationshipFieldMetadata,
+  isBBReferenceField,
+  isRelationshipField,
+  LinkDocument,
+  RelationshipFieldMetadata,
   RelationshipType,
   Row,
   Table,
-  isBBReferenceField,
-  isRelationshipField,
 } from "@budibase/types"
 import sdk from "../../../sdk"
-import { isExternalTable } from "../../../../src/integrations/utils"
-import { db as dbCore } from "@budibase/backend-core"
-import { EventType, updateLinks } from "../../../../src/db/linkedRows"
+import { isExternalTable } from "../../../integrations/utils"
+import { EventType, updateLinks } from "../../../db/linkedRows"
 import { cloneDeep } from "lodash"
 
+export interface MigrationResult {
+  tablesUpdated: Table[]
+}
+
 export async function migrate(
   table: Table,
   oldColumn: FieldSchema,
   newColumn: FieldSchema
-) {
-  let migrator = getColumnMigrator(table, oldColumn, newColumn)
-  let oldTable = cloneDeep(table)
+): Promise<MigrationResult> {
+  if (newColumn.name in table.schema) {
+    throw new BadRequestError(`Column "${newColumn.name}" already exists`)
+  }
 
   table.schema[newColumn.name] = newColumn
   table = await sdk.tables.saveTable(table)
 
-  await migrator.doMigration()
+  let migrator = getColumnMigrator(table, oldColumn, newColumn)
+  let result: MigrationResult
```

Comment:
very minor - in this sort of scenario, we would generally stick to:
```ts
try {
  return (await await sdk.tables.saveTable(table)) as MigrationResult
} catch (e) {
  // If the migration fails then we need to roll back the table schema
  // change.
  delete table.schema[newColumn.name]
  await sdk.tables.saveTable(table)
  throw e
}
```
This avoids the result variable being slightly out of scope and makes the typing a bit easier to understand.

Reply:
Ahh I think this was leftover from a previous iteration of this function that was trying to do something with `result` after the try-catch. Good spot! 🙏